src/Plugin/Notifier/Email.php

```diff
@@ -71,6 +72,13 @@ public static function create(ContainerInterface $container, array $configuratio
   public function deliver(array $output = []) {
     /** @var \Drupal\user\UserInterface $account */
     $account = $this->message->uid->entity;
+
+    if (!$this->configuration['mail'] && !$account->id()) {
```

Comment:
I think we do - it checks for the specific case where there's no user, and no `mail`. If we wait for getting mail below from the anon user, it throws an exception, but that is harder to understand.